pennylane/beta/plugins/default_tensor.py

```diff
@@ -400,36 +444,37 @@ def _add_node(self, A, wires, name="UnnamedNode"):
             node = A
         else:
             node = tn.Node(A, name=name, backend=self.backend)
-        self._nodes.append(node)
+
+        if key not in self._nodes:
+            self._nodes[key] = []
+        self._nodes[key].append(node)
+
         return node
 
     def _add_edge(self, node1, idx1, node2, idx2):
-        """Adds an edge to the underlying tensor network.
-
-        The edge is also added to ``self._edges`` for bookkeeping.
+        """Adds an edge between two nodes.
 
         Args:
             node1 (tn.Node): first node to connect
-            idx1 (int): index of node1 to add the edge to
+            idx1 (int): index of node1 to connect the edge to
             node2 (tn.Node): second node to connect
-            idx2 (int): index of node2 to add the edge to
+            idx2 (int): index of node2 to connect the edge to
 
         Returns:
             tn.Edge: the newly created edge
         """
         edge = tn.connect(node1[idx1], node2[idx2])
-        self._edges.append(edge)
-
         return edge
 
-    def pre_apply(self):
-        self.reset()
-
     def apply(self, operation, wires, par):
         if operation == "QubitStateVector":
             state = self._array(par[0], dtype=self.C_DTYPE)
             if state.ndim == 1 and state.shape[0] == 2 ** self.num_wires:
-                self._state_node.tensor = self._reshape(state, [2] * self.num_wires)
+                tensor = self._reshape(state, [2] * self.num_wires)
```

Comment:
(Suggestion for the method): as this method seems to be subject to growing, it might be worth breaking it up into auxiliary methods: e.g.:
```python
        if operation == "QubitStateVector":
                self.apply_qubit_state_vector(wires, par)
        # rest of apply()
```

Reply:
Thanks for suggestion. To save later effort, I think I will file this away for now until we decide if/how we want to port to `QubitDevice`

Reply:
That's a good call, as it might save some time indeed!